### PR TITLE
Add the my team settings page

### PR DESCRIPTION
**Pull request by acrantel**
Adds the team settings page. This PR doesn't add any functionality - only the design.
TODO:
- [x] profile section
- [x] elgibility and scrimmages sections
  - [x] fancy checkboxes (DescriptiveCheckbox component)
- [x] members list
- [x] no team? display team join stub 
- [x] leave team button

image

### CODE REVIEW

File: frontend2/src/components/team/MemberList.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      </div>
+    </div>
+  );
+};
+
+// Displays a list of the users in members. If the current user is in members,
+// displays the current user first.
+const MemberList: React.FC<MemberListProps> = ({ members, className = "" }) => {
+  const { user: currentUser } = useCurrentUser();
+  console.log(members[0]);
```
Comment (lowtorola) [inline frontend2/src/components/team/MemberList.tsx:41]:
```suggestion

```

No mo prints in console! 😃

File: frontend2/src/views/MyTeam.tsx [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+                />
+                <div className="text-center text-xl font-semibold">
+                  {team.name}
+                </div>
+              </div>
+              <div className="flex flex-1 flex-col gap-4">
+                <Input
+                  disabled
+                  label="Join key"
+                  value={team.join_key ?? "Loading..."}
```
Comment (lowtorola) [inline frontend2/src/views/MyTeam.tsx:46]: Should we add a spinner here?

File: frontend2/src/views/MyTeam.tsx [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+                <p className="text-gray-700">
+                  This determines which tournaments and prizes your team is
+                  eligible for.
+                </p>
+              </div>
+              <div className="flex flex-1 flex-col gap-2">
+                <DescriptiveCheckbox
+                  checked={checked}
+                  onChange={(checked) => {
+                    console.log("new value", checked);
```
Comment (lowtorola) [inline frontend2/src/views/MyTeam.tsx:78]:
```suggestion

```

Comment (lowtorola) [reply frontend2/src/views/MyTeam.tsx:78]: There's a few more of these below as well haha

**Review by lowtorola — CHANGES_REQUESTED** [review]
Left a few comments for your perusal, otherwise looks good!

File: frontend2/src/views/MyTeam.tsx (hunk 2 above)
Comment (acrantel) [reply frontend2/src/views/MyTeam.tsx:46]: b/c it's an input field I'm hesitant to add a spinner; we may want to consider making the whole page a spinner while the team is loading instead, but that can be added later

**Review by lowtorola — APPROVED** [review]
LGTM! ⭐

**Pull request merged**